openupgrade_scripts/scripts/base/16.0.1.3/post-migration.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2023 Hunki Enterprises BV (https://hunki-enterprises.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade, openupgrade_160
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.load_data(env.cr, "base", "16.0.1.3/noupdate_changes.xml")
+
+    if openupgrade.table_exists(
+        env.cr, "ir_translation"
+    ) and not openupgrade.table_exists(env.cr, "_ir_translation"):
+        openupgrade.rename_tables(env.cr, [("ir_translation", "_ir_translation")])
+
+    openupgrade_160.migrate_translations_to_jsonb(
```

Comment:
Rather than on a module by module basis, can we do this for all translatable fields in the database? I see the lib method uses the models from the environment, so that would have to be addressed.

Reply:
this would mean we have to convert columns to jsonb manually, and we'd also not be able to use https://github.com/OCA/OCB/blob/16.0/odoo/tools/translate.py#L1551 because this also expects the registry to be loaded for the field in question.

Or do you mean to move this to end-migration and do it in bulk there?

Reply:
The second one as we talked in OCA day :wink:

Reply:
Well, if it was the end-migration the process would not be applied to modules that are not loaded at that moment, such as when a setup is migrated from 15 to 17 and some modules are not provided for 16.0. So I would prefer here using our own SQL logic (also for performance).

Reply:
OK, I see, but I'm not sure if we can deduce which SQL columns to convert.

Reply:
I think we can go by the value in the `translate` column in `ir_model_fields`.

Reply:
Oh, I must admit that an exception are the fields with a `callable` translate value. This is not available for fields that are not loaded in the environment (at the end stage).

Reply:
I've implemented that in https://github.com/OCA/OpenUpgrade/pull/3680/commits/25830ead71b0183c91c7e486ad9295bcca95a6b2

Reply:
Brilliant, thanks! Shall I recreate a new version of the template database and install an additional language in it? We can then add a test :heart_eyes:
Do you plan a special treatment in the end-migration for fields that have a callable value for `translate` (once the field is loaded in the environment)?

Reply:
I actually didn't think about translate=callable yet, I think that can be a later problem.

The DB with an additional language would be very welcome indeed.